src/effects/native/flangereffect.cpp

```diff
@@ -28,26 +31,60 @@ EffectManifest FlangerEffect::getManifest() {
     period->setId("period");
     period->setName(QObject::tr("Period"));
     period->setDescription(QObject::tr("Controls the period of the LFO (low frequency oscillator)\n"
-        "1/4 - 4 beats rounded to 1/2 beat if tempo is detected (decks and samplers) \n"
-        "0.05 - 4 seconds if no tempo is detected (mic & aux inputs, master mix)"));
+        "1/4 - 20 beats rounded to 1/2 beat if tempo is detected (decks and samplers) \n"
+        "0.05 - 20 seconds if no tempo is detected (mic & aux inputs, master mix)"));
     period->setControlHint(EffectManifestParameter::ControlHint::KNOB_LINEAR);
     period->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
     period->setUnitsHint(EffectManifestParameter::UnitsHint::BEATS);
     period->setMinimum(0.00);
-    period->setMaximum(4.00);
+    period->setMaximum(20.00);
     period->setDefault(1.00);
 
-    EffectManifestParameter* depth = manifest.addParameter();
-    depth->setId("depth");
-    depth->setName(QObject::tr("Depth"));
-    depth->setDescription(QObject::tr("Controls the intensity of the effect."));
-    depth->setControlHint(EffectManifestParameter::ControlHint::KNOB_LINEAR);
-    depth->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
-    depth->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
-    depth->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
-    depth->setDefault(1.0);
-    depth->setMinimum(0.0);
-    depth->setMaximum(1.0);
+    EffectManifestParameter* width = manifest.addParameter();
+    width->setId("width");
+    width->setName(QObject::tr("Width"));
+    width->setDescription(QObject::tr("Controls the delay amplitude of the LFO (low frequency oscillator)."));
+    width->setControlHint(EffectManifestParameter::ControlHint::KNOB_LINEAR);
+    width->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    width->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    width->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    width->setDefault(kMaxLfoWidthMs);
+    width->setMinimum(0.0);
+    width->setMaximum(kMaxLfoWidthMs);
+
+    EffectManifestParameter* manual = manifest.addParameter();
+    manual->setId("manual");
+    manual->setName(QObject::tr("Manual"));
+    manual->setDescription(QObject::tr("Controls the delay offset of the LFO (low frequency oscillator).\n"
+            "With width at zero, it allows to manual sweep over the entire delay range."));
+    manual->setControlHint(EffectManifestParameter::ControlHint::KNOB_LINEAR);
+    manual->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    manual->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    manual->setDefault(kCenterDelayMs);
+    manual->setMinimum(kMinDelayMs);
+    manual->setMaximum(kMaxDelayMs);
+
+    EffectManifestParameter* regen = manifest.addParameter();
+    regen->setId("regen");
+    regen->setName(QObject::tr("Regen."));
```

Comment:
How about renaming this to "Feedback"?

This new parameter is really fun, thanks! :)

Reply:
Regen is the established name for that on hardware flangers.

Reply:
Okay, then set the full name to "Regeneration" and use "Regen" (no ".") for the shortName.